### PR TITLE
Delete Vertical Lanes

### LINKED ISSUE #2081: Vertical lane removal leads to unexpected outcome
A challenge to solve in the future:

https://github.com/bpmn-io/bpmn-js/assets/28307541/21ddc9a0-3c0a-450f-a306-76d85fcc14dd

_Originally posted by @barmac in https://github.com/bpmn-io/bpmn-js/pull/2067#pullrequestreview-1833154711_

### PR DESCRIPTION
**Pull request by sombrek**
Closes #2081

This is a small independent fix.

### CODE REVIEW

**Comment by sombrek** [comment]
It seems there is some kind of rounding issue when deleting lanes that causes their boundaries to mismatch. This affects both horizontal and vertical lanes. This can probably be fixed in the future.

delete-off

**Comment by barmac** [comment]
> It seems there is some kind of rounding issue when deleting lanes that causes their boundaries to mismatch. This affects both horizontal and vertical lanes. This can probably be fixed in the future.
> 
> delete-off [ delete-off ](https://private-user-images.githubusercontent.com/150448993/298374022-10aea9be-940f-44f2-ac75-f68a16283e89.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU5MTMwMjYsIm5iZiI6MTcwNTkxMjcyNiwicGF0aCI6Ii8xNTA0NDg5OTMvMjk4Mzc0MDIyLTEwYWVhOWJlLTk0MGYtNDRmMi1hYzc1LWY2OGExNjI4M2U4OS5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDEyMlQwODM4NDZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iMWRlZGZhMjM0NDM2YjU3YWU5NTViMDNhYTMyZDRlODgwNDlkMGVkNDdkMTQ2NTQxMzQ3MTdiZTE1NWI4ZTg2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.w7GA_pY1XqBZ1rt1Zw9q80k0N-29XaUdFVp1b0s3W3A) [ ](https://private-user-images.githubusercontent.com/150448993/298374022-10aea9be-940f-44f2-ac75-f68a16283e89.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU5MTMwMjYsIm5iZiI6MTcwNTkxMjcyNiwicGF0aCI6Ii8xNTA0NDg5OTMvMjk4Mzc0MDIyLTEwYWVhOWJlLTk0MGYtNDRmMi1hYzc1LWY2OGExNjI4M2U4OS5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDEyMlQwODM4NDZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iMWRlZGZhMjM0NDM2YjU3YWU5NTViMDNhYTMyZDRlODgwNDlkMGVkNDdkMTQ2NTQxMzQ3MTdiZTE1NWI4ZTg2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.w7GA_pY1XqBZ1rt1Zw9q80k0N-29XaUdFVp1b0s3W3A)

Could you please share a diagram in which you got this result? I cannot reproduce it on my own.

**Comment by sombrek** [comment]
> Could you please share a diagram in which you got this result? I cannot reproduce it on my own.

Sure, it's [test/spec/features/modeling/lanes/lanes.bpmn](https://github.com/bpmn-io/bpmn-js/blob/develop/test/spec/features/modeling/lanes/lanes.bpmn)
Open it via bpmn.io then delete lane c and lane b.

**Comment by barmac** [comment]
Thank you! I can reproduce it now. We should make sure that the child lanes don't exceed the parent lane bounds.

**Comment by barmac** [comment]
I moved this to a separate issue as it's not required for this PR: https://github.com/bpmn-io/bpmn-js/issues/2083

File: lib/features/modeling/behavior/DeleteLaneBehavior.js [hunk 1]
```diff
@@ -106,7 +150,9 @@ export default function DeleteLaneBehavior(eventBus, spaceTool) {
       return;
     }
 
-    compensateLaneDelete(shape, oldParent);
+    var isHorizontalLane = isHorizontal(shape);
+
+    compensateLaneDelete(shape, oldParent, isHorizontalLane);
```
Comment (barmac) [inline lib/features/modeling/behavior/DeleteLaneBehavior.js:155]: I'd rather move the `isHorizontal` call to the inside of `compensateLaneDelete`. There's no case where we'd pass `isHorizontalLane=false` for horizontal lanes to this function.

Comment (barmac) [reply lib/features/modeling/behavior/DeleteLaneBehavior.js:155]: I will do that as a quick change.

**Comment by barmac** [comment]
> Thank you! I can reproduce it now. We should make sure that the child lanes don't exceed the parent lane bounds.

BTW this bug can be noticed on the vertical lanes fixture that you're adding in this PR.

**Comment by barmac** [comment]
There is a width missmatch in the DI:

```
      <bpmndi:BPMNShape id="BPMNShape_140jwaa" bpmnElement="Vertical_Participant_Lane" isHorizontal="false">
        <dc:Bounds x="160" y="50" width="603" height="540" />
        <bpmndi:BPMNLabel />
      </bpmndi:BPMNShape>
      <bpmndi:BPMNShape id="BPMNShape_0n0ckvg" bpmnElement="Vertical_Lane_B" isHorizontal="false">
        <dc:Bounds x="655" y="80" width="108" height="510" />
        <bpmndi:BPMNLabel />
      </bpmndi:BPMNShape>
      <bpmndi:BPMNShape id="BPMNShape_1xhgbhc" bpmnElement="Vertical_Lane_A" isHorizontal="false">
        <dc:Bounds x="160" y="80" width="496" height="510" />
        <bpmndi:BPMNLabel />
      </bpmndi:BPMNShape>
```

Notice that the child lanes' widths sum up to 604 while the parent is 603.

**Comment by barmac** [comment]
Thank you @sombrek for your another great contribution!

**Pull request merged**